ocaml/typing/typecore.ml

```diff
@@ -5435,28 +5435,30 @@ and type_expect_
         exp_env = env;
       }
   | Pexp_letop{ let_ = slet; ands = sands; body = sbody } ->
-      let rec loop spat_acc ty_acc sands =
+      let rec loop spat_acc ty_acc ty_acc_sort sands =
         match sands with
-        | [] -> spat_acc, ty_acc
+        | [] -> spat_acc, ty_acc, ty_acc_sort
         | { pbop_pat = spat; _} :: rest ->
             (* CR layouts v5: eliminate value requirement *)
             let ty = newvar (Layout.value ~why:Tuple_element) in
             let loc = Location.ghostify slet.pbop_op.loc in
             let spat_acc = Ast_helper.Pat.tuple ~loc [spat_acc; spat] in
             let ty_acc = newty (Ttuple [ty_acc; ty]) in
-            loop spat_acc ty_acc rest
+            loop spat_acc ty_acc Sort.value rest
```

Comment:
`Sort.for_tuple`, perhaps?

Even better would be `Sort.for_letop_tuple`, anticipating the possibility of using unboxed tuples here (which we really should). But the odds of missing changing this sort when we do that are pretty slim.